src/main/java/aquality/selenium/core/elements/interfaces/IParent.java

```diff
@@ -8,31 +8,100 @@
     /**
      * Find an element in the parent element
      *
-     * @param childLoc Child element locator
-     * @param clazz class or interface of the element to be obtained
+     * @param childLoc child element locator
+     * @param name     output name in logs
+     * @param clazz    class or interface of the element to be obtained
+     * @param state    visibility state of target element
+     * @param <T>      the type of the element to be obtained
      * @return found child element
      */
-    default <T extends IElement> T findChildElement(By childLoc, Class<? extends IElement> clazz){
-        return findChildElement(childLoc, clazz, ElementState.DISPLAYED);
+    <T extends IElement> T findChildElement(By childLoc, String name, Class<T> clazz, ElementState state);
+
+    /**
+     * Find an element in the parent element with DISPLAYED state
+     *
+     * @param childLoc child element locator
+     * @param name     output name in logs
+     * @param clazz    class or interface of the element to be obtained
+     * @param <T>      the type of the element to be obtained
+     * @return found child element
+     */
+    default <T extends IElement> T findChildElement(By childLoc, String name, Class<T> clazz) {
+        return findChildElement(childLoc, name, clazz, ElementState.DISPLAYED);
     }
 
     /**
      * Find an element in the parent element
      *
-     * @param childLoc Child element locator
-     * @param clazz class or interface of the element to be obtained
-     * @param state visibility state of target element
+     * @param childLoc child element locator
+     * @param clazz    class or interface of the element to be obtained
+     * @param state    visibility state of target element
+     * @param <T>      the type of the element to be obtained
      * @return found child element
      */
-    <T extends IElement> T findChildElement(By childLoc, Class<? extends IElement> clazz, ElementState state);
+    default <T extends IElement> T findChildElement(By childLoc, Class<T> clazz, ElementState state) {
+        return findChildElement(childLoc, null, clazz, state);
```

Comment:
I think we can forbid to create elements without names, can't we? 
So, I suggest removing such signatures 
here and in the method below

Reply:
discussed